### ISSUE TITLE
could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED

### ISSUE THREAD

**Issue by kaz94** [tensorflow/tensorflow#30254]
Hello,
For a long time now, I've been trying to resolve problem from the header:
could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
I went through all of the similar issues but none of the solutions worked for me.

this log: possibly insufficient driver version: 387.26.0
seems to indicate that I have a bad driver version but for CUDA 9.0 documentation (https://docs.nvidia.com/deploy/cuda-compatibility/index.html#binary-compatibility__table-toolkit-driver) says:
CUDA 9.0 (9.0.76)	>= 384.81
which is lower than I currently have installed (387.26)
I really hope to get some help,
cheers, Kasia

**System information**
Running tensorflow inside singularity container:
https://singularity-hub.org/collections/3092
singularity pull --name eric.img shub://Eric716/tensorflow-gpu-cuda9.0:fullversion
singularity shell --nv eric.img

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
EMI release 3.0 (Monte Bianco)
LSB_VERSION=base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Scientific Linux CERN SLC release 6.5 (Carbon)

- TensorFlow installed from (source or binary):
- TensorFlow version (use command below): tensorflow-gpu      1.8.0
Keras               2.2.4
- Python version: Python 3.5.2
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA Version 9.0.176 / 
cuDNN: 
cat /usr/include/cudnn.h | grep CUDNN_MAJOR -A 2
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 4
#define CUDNN_PATCHLEVEL 2

nvidia-smi
Driver Version: 387.26
- GPU model and memory:

**Describe the current behavior**
Using TensorFlow backend.
2019-06-30 00:10:58.871234: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties: 
name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235
pciBusID: 0000:0d:00.0
totalMemory: 11.17GiB freeMemory: 11.10GiB
2019-06-30 00:10:58.871309: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2019-06-30 00:10:59.318801: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-06-30 00:10:59.318875: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0 
2019-06-30 00:10:59.318897: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N 
2019-06-30 00:10:59.320282: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8007 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:0d:00.0, compute capability: 3.7)
2019-06-30 00:11:00.116849: E tensorflow/stream_executor/cuda/cuda_dnn.cc:455] could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2019-06-30 00:11:00.380252: E tensorflow/stream_executor/cuda/cuda_dnn.cc:463] possibly insufficient driver version: 387.26.0
2019-06-30 00:11:00.380342: F tensorflow/core/kernels/conv_ops.cc:713] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms) 
Aborted

**Code to reproduce the issue**
import numpy as np
from keras.models import Input, Model
from keras.layers import Conv1D
from keras.optimizers import *
import keras.backend as K

def conv_model(input_size=(256, 1)):
    cfg = K.tf.ConfigProto(gpu_options={'allow_growth': True, 'per_process_gpu_memory_fraction': 0.7})
    K.set_session(K.tf.Session(config=cfg))

    inputs = Input(input_size)
    conv = Conv1D(1, 3, padding='same')(inputs)
    model = Model(inputs=inputs, outputs=conv)
    model.compile(optimizer=Adam(lr=2e-4), loss='binary_crossentropy', metrics=['accuracy'])
    return model

if __name__ == '__main__':
    num_epochs = 300
    model = conv_model()
    input_batch = np.array([np.concatenate([[[0]]*100, [[1]]*56, [[0]]*100])])

    for epoch in range(num_epochs):
        loss, acc = model.train_on_batch(input_batch, input_batch)
        print('loss: {}, acc: {}'.format(loss, acc))

**Comment by ravikyram** [comment]
Please, let us know which version of cuDNN you installed?If not please install cuDNN 7 and let us know if the issue still persists.Thanks!

**Comment by kaz94** [comment]
@ravikyram 
from this output:
cat /usr/include/cudnn.h | grep CUDNN_MAJOR -A 2
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 4
#define CUDNN_PATCHLEVEL 2
I assume it's 7.4.2 am I right?

**Comment by jvishnuvardhan** [comment]
@kaz94 I don't see any issue with the driver and also your code. Source of the issue is something else.

Did you install tensorflow from source?
Did you check whether there is any issue with the driver 387.26? 
Did you try upgrading or downgrading the version of driver (example downgrading to 384.81)
Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by vinaykumar2491** [comment]
> @kaz94 I don't see any issue with the driver and also your code. Source of the issue is something else.
> 
> Did you install tensorflow from source?
> Did you check whether there is any issue with the driver 387.26?
> Did you try upgrading or downgrading the version of driver (example downgrading to 384.81)
> Thanks!

I have the same issue with `tensorflow-gpu==1.8.0`, `cudnn==7.0.5` and `cuda=9.1.84` on ubuntu 16.04
Below is the log:

Hi everyone,
I am trying to reproduce the results mentioned in paper  https://github.com/f90/Wave-U-Net ; but when I run the repo's scripts I get the following errors:
```
2019-07-25 05:51:38.470459: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:94] Filling up shuffle buffer (this may take a while): 2001 of 8000
2019-07-25 05:51:48.423484: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:94] Filling up shuffle buffer (this may take a while): 4201 of 8000
2019-07-25 05:51:58.262603: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:94] Filling up shuffle buffer (this may take a while): 6448 of 8000
2019-07-25 05:52:04.644012: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:129] Shuffle buffer filled.
2019-07-25 05:52:04.890160: E tensorflow/stream_executor/cuda/cuda_dnn.cc:455] could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2019-07-25 05:52:04.890268: E tensorflow/stream_executor/cuda/cuda_dnn.cc:463] possibly insufficient driver version: 384.183.0
2019-07-25 05:52:04.890282: E tensorflow/stream_executor/cuda/cuda_dnn.cc:427] could not destroy cudnn handle: CUDNN_STATUS_BAD_PARAM
2019-07-25 05:52:04.890291: F tensorflow/core/kernels/conv_ops.cc:713] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parameters.ShouldIncludeWino
gradNonfusedAlgo<T>(), &algorithms) 
Aborted (core dumped)
```
Please help.
Thankyou

**Comment by jvishnuvardhan** [comment]
@vinaykumar2491 Please create a new issue with platform details, standalone code and any other details that will help in resolving the issue faster. Thanks!
